### PR TITLE
Map species

### PR DESCRIPTION
**Pull request by joefitter**
* Map species from training module to certificate level

### CODE REVIEW

File: lib/routers/profile/person.js [hunk 1]
```diff
@@ -100,6 +100,18 @@ router.get('/', (req, res, next) => {
...
     .then(profile => {
+      if (profile && profile.certificates) {
+        profile.certificates = profile.certificates.map(cert => {
+          if (cert.species && cert.species.length) {
+            return cert;
+          }
+          return {
+            ...cert,
+            species: (cert.modules || [])
+              .reduce((arr, module) => [ ...arr, ...(module.species || []) ], [])
```
Comment (wheelsandcogs) [inline lib/routers/profile/person.js]: can we extract this reduce function and give it a meaningful name please?

Comment (joefitter) [reply lib/routers/profile/person.js]: for you, anything x

File: lib/routers/profile/person.js [hunk 2]
```diff
@@ -96,10 +96,24 @@ const getSingleProfile = req => {
...
 const router = Router({ mergeParams: true });
 
+function mapSpeciesFromModules(cert) {
+  if (cert.species && cert.species.length) {
+    return cert;
+  }
+  return {
+    ...cert,
+    species: (cert.modules || [])
+      .reduce((arr, module) => [ ...arr, ...(module.species || []) ], [])
```
Comment (lennym) [inline lib/routers/profile/person.js]: Minor quibble, but avoid using `module` as a variable name as it's already a keyword.

Comment (joefitter) [reply lib/routers/profile/person.js]: ah that's a good point - I'll update

**Pull request merged**